packages/cli/src/workflow-runner.ts

```diff
@@ -286,12 +295,54 @@ export class WorkflowRunner {
 					data.executionData,
 				);
 				workflowExecution = workflowExecute.processRunExecutionData(workflow);
+			} else if (data.preferredTrigger?.data && data.startNodes && !data.destinationNode) {
+				this.logger.debug(
+					`Execution ID ${executionId} had preferredTrigger. Starting from that trigger.`,
+					{ executionId },
+				);
+				const startNodes = data.startNodes.map((data) => {
+					const node = workflow.getNode(data.name);
+					a.ok(node, `Could not find a node named "${data.name}" in the workflow.`);
+					return node;
+				});
+
+				const { nodeExecutionStack, waitingExecution, waitingExecutionSource } =
+					recreateNodeExecutionStack(
+						filterDisabledNodes(DirectedGraph.fromWorkflow(workflow)),
+						new Set(startNodes),
+						{ [data.preferredTrigger.name]: [data.preferredTrigger.data] },
+						data.pinData ?? {},
+					);
+				const executionData: IRunExecutionData = {
+					resultData: {
+						runData: data.runData ?? {
+							[data.preferredTrigger.name]: [data.preferredTrigger.data],
```

Comment:
Why don't we use the existing run data at L313 like we do here?

Reply:
You're right, we should not use the runData from the FE if we're told to start from a trigger.